### PR TITLE
Fix deletion in the overlapping connector case

### PR DESCRIPTION
**Pull request by Weves**
Wrote test, it failed. Fixed. It worked.

### CODE REVIEW

File: backend/danswer/db/document_set.py [hunk 1]
```diff
@@ -525,36 +525,52 @@ def fetch_document_sets_for_documents(
 ) -> Sequence[tuple[str, list[str]]]:
     """Gives back a list of (document_id, list[document_set_names]) tuples"""
     stmt = (
```
Comment (Weves) [inline backend/danswer/db/document_set.py:527]: Previously, this just wouldn't return a row at all for document that didn't have any document sets associated with them.

File: backend/danswer/db/document_set.py [hunk 2]
```diff
@@ -525,36 +525,52 @@ def fetch_document_sets_for_documents(
...
-            DocumentSet__ConnectorCredentialPair,
-            DocumentSetDBModel.id
-            == DocumentSet__ConnectorCredentialPair.document_set_id,
+        select(
+            Document.id,
+            func.coalesce(
+                func.array_remove(func.array_agg(DocumentSetDBModel.name), None), []
+            ).label("document_set_names"),
         )
-        .join(
```
Comment (Weves) [inline backend/danswer/db/document_set.py:534]: joins are jsut re-ordered by Claude :(

File: backend/danswer/server/manage/administrative.py [hunk 3]
```diff
@@ -166,10 +166,14 @@ def create_deletion_attempt_for_connector_id(
         get_editable=True,
     )
     if cc_pair is None:
+        error = (
```
Comment (Weves) [inline backend/danswer/server/manage/administrative.py:169]: adding more logging (previously would not give any details in the logs)

File: backend/danswer/background/update.py [hunk 4]
```diff
@@ -67,6 +68,10 @@ def _should_create_new_indexing(
 ) -> bool:
     connector = cc_pair.connector
 
+    # don't kick off indexing for `NOT_APPLICABLE`
```
Comment (Weves) [inline backend/danswer/background/update.py:71]: this was throwing off tests, should be a no-op in production

File: backend/danswer/db/document_set.py (hunk 1 above)
Comment (hagen-danswer) [reply backend/danswer/db/document_set.py:527]: outerjoin gang

File: backend/danswer/server/manage/administrative.py (hunk 3 above)
Comment (hagen-danswer) [reply backend/danswer/server/manage/administrative.py:169]: I have been doing the same thing. 

However, this is just a consequence of the recent logging rework causing failed API calls to not show any error beyond whatever 400 code. We should probably look into solving that...

**Review by hagen-danswer — APPROVED** [review]
lgtm

I will merge so i can rebase/refactor

File: backend/tests/integration/tests/connector/test_deletion.py [hunk 5]
```diff
@@ -188,3 +189,123 @@ def test_connector_deletion(reset: None, vespa_client: TestVespaClient) -> None:
...
+        if c1_details.connector_id not in [c["id"] for c in connectors]:
+            break
+
+        if time.time() - start > MAX_DELAY:
+            raise TimeoutError("Connector 1 was not deleted within the max delay")
+
+        time.sleep(2)
+
+    print("Connector 1 deleted")
+
```
Comment (hagen-danswer) [inline backend/tests/integration/tests/connector/test_deletion.py:292]: check that only connector 2 is deleted?

**Pull request closed without merging**